WonSeungHyun/app.js

```diff
@@ -37,3 +35,118 @@ const start = async () => {
 };
 
 start();
+
+app.post("/users", async (req, res, next) => {
+  const { name, email, profile_image } = req.body;
+
+  await appDataSource.query(
+    `
+    INSERT INTO users(
+    name,
+    email,
+    profile_image
+    
+  )VALUES (?, ?, ?)`,
+    [name, email, profile_image]
+  );
+  res.status(201).json({ message: "successfully created" });
+});
+
+app.get("/search", async (req, res) => {
+  await appDataSource.query(
+    `SELECT
+     u.id as userId,
+     u.profile_image as userProfileImage,
+     p.id as PostingId,
+     
+     p.content as PostingContent
+     FROM posts p, users u
+     WHERE p.user_id = u.id`,
+    (err, rows) => {
+      res.status(200).json(rows);
+    }
+  );
+});
+////유저가 작성한 게시물 불러오기 기능///
+app.get("/search_user", async (req, res) => {
+  await appDataSource.query(
+    `SELECT
+    u.id as userId,
+    u.profile_image as userProfileImage,
+    p.id as postingId,
+    p.posting_img as postingImageUrl,
+    p.content as postingContent
+    FROM users u , posts p
+    WHERE p.user_id = u.id
+    
+    `,
+    (err, rows) => {
+      res.status(200).json(rows);
+    }
+  );
+});
+////특정 유저 게시물 조회 기능///
+app.get("/userposts/:inputId", async (req, res) => {
+  const userId = req.params.inputId;
+  const user = await appDataSource.manager.query(
+    `SELECT
+          users.id as userId,
+          users.profile_image as userProfileImage
+      FROM users
+      WHERE users.id = ${userId};
+      `
+  );
+  const userpost = await appDataSource.manager.query(
+    `SELECT 
+          posts.id as postingId,
+          posts.posting_img as postingImageUrl,
+          posts.content as postingContent
+      FROM posts
+      WHERE user_id = ${userId};`
+  );
+
+  user[0].posting = userpost;
+  res.status(200).json({ data: user[0] });
+});
```

Comment:
path parameter를 활용하여 특정 데이터를 조회하는 방법에 대해 잘 작성해주셨습니다.

아래 수정사항을 참고하여 해당 부분도 수정해주세요!
```suggestion
app.get("/posts/:userId", async (req, res) => {
  const { userId } = req.params;
  const  posts = await appDataSource.query(`
    SELECT
      u.id as userId,
      u.profile_image as userProfileImage,
      p.id as postingId,
      p.title as postingTitle,
      p.content as postingContent
    FROM posts as p
    INNER JOIN users as u ON u.id = p.user_id
    WHERE p.user_id = ? 
  `, [ userId ])
  res.status(200).json({ data: post });
});
```